packages/shared/friends/sagas.ts

```diff
@@ -2028,6 +2066,93 @@ export async function getChannelMembers(request: GetChannelMembersPayload) {
   getUnityInstance().UpdateChannelMembers(channelMembersPayload)
 }
 
+export async function requestFriendship(request: SendFriendRequestPayload) {
+  try {
+    let userId = request.userId
+    let found = false
+    const state = store.getState()
+
+    const ownId = getOwnId(state)
+
+    if (!ownId) {
+      return { reply: null, error: FriendshipErrorCode.FEC_UNKNOWN }
+    }
+
+    // Search user profile on server
+    if (isAddress(userId)) {
+      // Ensure user profile is initialized and send it to renderer
+      const avatar = await ensureFriendProfile(userId)
+      found = avatar.hasConnectedWeb3 || false
+    } else {
+      const profileByName = findProfileByName(state, userId)
+      if (profileByName) {
+        userId = profileByName.userId
+        found = true
+      }
+    }
+
+    // If user profile was not found on the server -> no connected web3, check if it has a name claimed
+    if (!found) {
+      const net = getSelectedNetwork(state)
+      const address = await fetchENSOwner(ethereumConfigurations[net].names, userId)
+      if (address) {
+        // If an address was found by the name, set it as user id
+        userId = address
+        found = true
+      }
+    }
+
+    if (!found) {
+      return { reply: null, error: FriendshipErrorCode.FEC_NON_EXISTING_USER }
+    }
+
+    // Update user data
+    store.dispatch(updateUserData(request.userId.toLowerCase(), getMatrixIdFromUser(request.userId)))
```

Comment:
Shouldn't it be using the `userId` local variable instead of the request one here?

Reply:
Actually, we're not requesting the user id, we're only formatting it. Should I change the name to something more telling and less confusing? Do I do it now? What you think?

```
Get the matrixUserId from userId

@param userId — a string with the userId pattern

@example

from: '0x1111ada11111'
to: '@0x1111ada11111:decentraland.org'
```

Reply:
Never mind— I get it. You're right. Thx